validator_worker/src/follower.rs

```diff
@@ -333,3 +333,527 @@ async fn on_error<'a, C: Unlocked + 'static>(
+            };
+            let res = on_new_state(
+                &sentry,
+                &channel_context,
+                get_initial_balances(),
+                new_state,
+                UnifiedNum::from_u64(4000),
+            )
+            .await
+            .expect("Shouldn't return an error");
+            assert!(matches!(
+                res,
+                ApproveStateResult::RejectedState {
+                    reason: InvalidNewState::Transition,
+                    ..
+                }
+            ), "InvalidNewState::Transition is the rejection reason when proposed balances are lower than the previous balances");
+        }
+
+        // - Case where get_health() will return less than 750 promilles
+        {
+            let balances = get_initial_balances();
+            let state_root = balances
+                .encode(
+                    channel_context.context.id(),
+                    channel_context.token.precision.get(),
+                )
+                .expect("Should encode");
+            let _mock_guard = server_setup
+                .setup_last_approved_response(balances.into_unchecked(), state_root)
+                .await;
+
+            let mut our_balances = get_initial_balances();
+            our_balances
+                .spend(*ADVERTISER, *PUBLISHER, UnifiedNum::from_u64(200_000))
+                .expect("should spend");
+            our_balances
+                .spend(*GUARDIAN, *PUBLISHER_2, UnifiedNum::from_u64(200_000))
+                .expect("should spend");
+
+            let state_root = get_initial_balances()
+                .encode(
+                    channel_context.context.id(),
+                    channel_context.token.precision.get(),
+                )
+                .expect("should encode");
+            let new_state: NewState<UncheckedState> = NewState {
+                state_root: state_root,
+                signature: IDS[&*LEADER].to_checksum(),
+                balances: get_initial_balances().into_unchecked(),
+            };
+            let res = on_new_state(
+                &sentry,
+                &channel_context,
+                our_balances,
+                new_state,
+                UnifiedNum::from_u64(1_000_000),
+            )
+            .await
+            .expect("Shouldn't return an error");
+            assert!(matches!(
+                res,
+                ApproveStateResult::RejectedState {
+                    reason: InvalidNewState::Health(..),
+                    ..
+                }
+            ), "InvalidNewState::Health is the rejection reason when the health penalty is too high");
+        }
+
+        // Case where no NewState is returned
+        {
+            // Setting up the expected response
+            let _mock_guard = server_setup.setup_new_state_response(None).await;
+            let _mock_guard = server_setup.setup_approve_state_response(None).await;
```

Comment:
The `ServerSetup` functions have an `.expect(..)` which asserts the exact number of calls. We can pass it as a function argument for better clarity.

Reply:
Ahhh, yes, this is good then. My only question remaining is if the mock_guard is actually working.
Wiremock says that `When expectations of a scoped [Mock](https://docs.rs/wiremock/latest/wiremock/struct.Mock.html) are not verified, it will trigger a panic` but I can't be certain right now that shadowing the variable like this will actually trigger the drop.
Can we verofy this?